lightning/src/ln/channelmanager.rs

```diff
@@ -3784,26 +3797,30 @@ impl<Signer: Sign, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelMana
 	/// Provides a payment preimage in response to [`Event::PaymentReceived`], generating any
 	/// [`MessageSendEvent`]s needed to claim the payment.
 	///
+	/// Note that calling this method does *not* guarantee that the payment has been claimed. You
+	/// *must* wait for an [`Event::PaymentClaimed`] event. In most cases, this event will be
+	/// generated before this function returns and will be provided to your [`EventHandler`] when
+	/// [`process_pending_events`] is next called.
```

Comment:
>  In most cases, this event will be generated before this function returns

Is there a case where we expect the event to be generated *only* after calling `claim_funds` but it is not generated before the function returns? Saying "most cases" seems to imply there is such a case.

Reply:
Ah, ok. How about simply:

> You *must* wait for an [`Event::PaymentClaimed`] event, which upon a success claim would be provided to your [`EventHandler`] when [`process_pending_events`] is next called.

Reply:
Also needs to be updated where events are created.